core/kernel/persistence/starsql/class.Class.php

```diff
@@ -309,8 +404,168 @@ public function deleteInstances(core_kernel_classes_Class $resource, $resources,
+            if ($orderProperty->isLgDependent()) {
+                $predicate = $this->buildLanguagePattern($predicate->toQuery(), $lang);
+            }
+            //Can't use dedicated order function as it doesn't support raw expressions
+            $query->raw(
+                'ORDER BY',
+                $predicate->toQuery() . ((strtoupper($orderDir) === 'DESC') ? ' DESCENDING': '')
+            );
+        }
+
+        if ($limit > 0) {
+            $query
+                ->skip($offset)
+                ->limit($limit);
+        }
+
+        return $query->build();
+    }
+
+    private function buildPropertyQuery(
+        $predicate,
+        $values,
+        bool $like
+    ): BooleanType {
+        if (!is_array($values)) {
+            $values = [$values];
+        }
+
+        $valuePatterns = null;
+        $lastItem = array_key_last($values);
+        foreach ($values as $key => $val) {
+            if ($val instanceof core_kernel_classes_Resource) {
+                $returnValue = $predicate->equals($val->getUri());
+            } else {
+                $patternToken = trim((string)$val);
+                if ($like) {
+                    $isWildcard = (strpos($patternToken, '*') !== false);
+                    $patternToken = strtr(
+                        trim($patternToken, '%'),
+                        [
+                            '.' => '\\.',
+                            '\_' => '_',
+                            '\%' => '%',
+                            '*' => '.*',
+                            '_' => '.',
+                            '%' => '.*',
+                        ]
+                    );
+                    if (!$isWildcard) {
+                        $patternToken = '.*' . $patternToken . '.*';
+                    }
+                    $returnValue = $predicate->regex('(?i)' . $patternToken);
+                } else {
+                    $returnValue = $predicate->equals($patternToken);
+                }
+            }
+
+            $valuePatterns = ($valuePatterns)
+                ? $valuePatterns->or($returnValue, ($key === $lastItem))
+                : $returnValue;
+        }
+
+        return $valuePatterns;
+    }
+
+    /**
+     * @param string $predicate
+     * @param string $lang
+     *
+     * @return RawExpression
+     */
+    private function buildLanguagePattern(string $predicate, string $lang = ''): RawExpression
+    {
+        $defaultLanguage = $this->getDefaultLanguage();
+
+        if (empty($lang) || $lang === $defaultLanguage) {
+            $resultExpression = Query::rawExpression(
+                sprintf(
+                    "n10s.rdf.getLangValue('%s', %s)",
+                    $defaultLanguage,
+                    $predicate
+                )
+            );
+        } else {
+            $resultExpression = Query::rawExpression(
+                sprintf(
+                    "coalesce(n10s.rdf.getLangValue('%s', %s), n10s.rdf.getLangValue('%s', %s))",
```

Comment:
Can we assume that language dependent properties will always have a language suffix, or should we have a fallback in case suffix is not there?

Reply:
This is a really good question. From what I see through the import, it always has some language, at least a default one. However, it will depend on the implementation we will do for neo4j. On the other hand, I prefer to have at least a default language translation as those properties are maps and we should always have a key in the map